src/metabase/api/user.clj

```diff
@@ -19,43 +19,42 @@
   (api/check-403 (or (= user-id api/*current-user-id*)
                      (:is_superuser @api/*current-user*))))
 
+(def ^:private all-user-fields
+  (vec (cons User user/all-user-fields)))
+
 (api/defendpoint GET "/"
-  "Fetch a list of all active `Users` for the admin People page."
+  "Fetch a list of all `Users` for the admin People page."
   []
-  (db/select [User :id :first_name :last_name :email :is_superuser :google_auth :ldap_auth :last_login]
-    :is_active true))
-
-(defn- reactivate-user! [existing-user first-name last-name]
-  (when-not (:is_active existing-user)
-    (db/update! User (u/get-id existing-user)
-      :first_name    first-name
-      :last_name     last-name
-      :is_active     true
-      :is_superuser  false
-      ;; if the user orignally logged in via Google Auth and it's no longer enabled, convert them into a regular user
-      ;; (see Issue #3323)
-      :google_auth   (boolean (and (:google_auth existing-user)
-                                   ;; if google-auth-client-id is set it means Google Auth is enabled
-                                   (session-api/google-auth-client-id)))
-      :ldap_auth     (boolean (and (:ldap_auth existing-user)
-                                   (ldap/ldap-configured?)))))
+  (db/select all-user-fields))
+
+(defn- fetch-user [& query-criteria]
+  (apply db/select-one all-user-fields query-criteria))
+
+(defn- reactivate-user! [existing-user]
+  (db/update! User (u/get-id existing-user)
+    :is_active     true
+    :is_superuser  false
+    ;; if the user orignally logged in via Google Auth and it's no longer enabled, convert them into a regular user
+    ;; (see Issue #3323)
+    :google_auth   (boolean (and (:google_auth existing-user)
+                                 ;; if google-auth-client-id is set it means Google Auth is enabled
+                                 (session-api/google-auth-client-id)))
+    :ldap_auth     (boolean (and (:ldap_auth existing-user)
+                                 (ldap/ldap-configured?))))
   ;; now return the existing user whether they were originally active or not
-  (User (u/get-id existing-user)))
+  (fetch-user :id (u/get-id existing-user)))
 
 
 (api/defendpoint POST "/"
-  "Create a new `User`, or reactivate an existing one."
+  "Create a new `User`, return a 400 if the email address is already taken"
   [:as {{:keys [first_name last_name email password]} :body}]
   {first_name su/NonBlankString
    last_name  su/NonBlankString
    email      su/Email}
   (api/check-superuser)
-  (if-let [existing-user (db/select-one [User :id :is_active :google_auth], :email email)]
-    ;; this user already exists but is inactive, so simply reactivate the account
-    (reactivate-user! existing-user first_name last_name)
-    ;; new user account, so create it
-    (user/invite-user! first_name last_name email password @api/*current-user*)))
-
+  (api/checkp (not (db/exists? User :email email))
```

Comment:
@camsaul I wasn't aware of https://github.com/metabase/metabase/pull/7481#discussion_r186896077. Am I using the functions correctly here to get our standard error messages?

Reply:
Yeah that looks right. I totally forgot this function existed.

Reply:
It was previously more of an informal convention than anything else, however the `defendpoint` param validation code all returns errors in that format so we might as well make it consistent and try to do it everywhere